src/backend/tests/projects.test.ts

```diff
@@ -257,6 +263,102 @@ describe('Projects', () => {
+      await expect(
+        async () =>
+          await ProjectsService.createAssembly('new assembly', 'file.txt', batman, {
+            carNumber: 1,
+            projectNumber: 1,
+            workPackageNumber: 0
+          })
+      ).rejects.toThrow(new NotFoundException('Project', '1.1.0'));
+    });
+
+    test('createAssembly fails when project has been deleted', async () => {
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue({
+        wbsElement: { ...prismaProject1.wbsElement, dateDeleted: new Date() },
+        projectId: prismaProject1.projectId,
+        teams: justiceLeague
+      } as any);
+      await expect(
+        async () =>
+          await ProjectsService.createAssembly('new assembly', 'file.txt', batman, {
+            carNumber: 1,
+            projectNumber: 1,
+            workPackageNumber: 0
+          })
+      ).rejects.toThrow(new DeletedException('Project', prismaProject1.projectId));
+    });
+
+    test('createAssembly fails when no permissions', async () => {
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue({
+        wbsElement: { ...prismaProject1.wbsElement, dateDeleted: '' },
+        projectId: prismaProject1.projectId
+      } as any);
+      await expect(
+        async () =>
+          await ProjectsService.createAssembly('new assembly', 'file.txt', wonderwoman, {
+            carNumber: 1,
+            projectNumber: 1,
+            workPackageNumber: 0
+          })
+      ).rejects.toThrow(new AccessDeniedException(`Users must be admin, or assigned to the team to create assemblies`));
+    });
+
+    test('createAssembly fails when no permissions - leadership', async () => {
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue({
+        wbsElement: { ...prismaProject1.wbsElement, dateDeleted: '' },
+        projectId: prismaProject1.projectId
+      } as any);
+      await expect(
+        async () =>
+          await ProjectsService.createAssembly('new assembly', 'file.txt', aquaman, {
+            carNumber: 1,
+            projectNumber: 1,
+            workPackageNumber: 0
+          })
+      ).rejects.toThrow(new AccessDeniedException(`Users must be admin, or assigned to the team to create assemblies`));
+    });
+
+    test('createAssembly works if the submitter is admin', async () => {
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue(prismaProject1);
+      vi.spyOn(prisma.assembly, 'create').mockResolvedValue(prismaAssembly1);
+
+      // no error, no return value
+      await ProjectsService.createAssembly('new assembly', 'file.txt', batman, {
+        carNumber: 1,
+        projectNumber: 1,
+        workPackageNumber: 0
+      });
+    });
+
+    test('createAssembly works if the submitter is on the team', async () => {
+      mockGetHighestProjectNumber.mockResolvedValue(0);
+      vi.spyOn(prisma.assembly, 'create').mockResolvedValue(prismaAssembly1);
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue(prismaProject1);
```

Comment:
for some reason this is reading the users on in the leads array on the team on the project as undefined, you can manually add them in using the spread operator like: `{...prismaProject1, teams: [{...prismaTeam1, leads:[]}]}`
the spread operator places all the attributes into the specified object into the newly created object, and lets you override any values like 'teams'.